src/lib/paymentOptions/NimiqPaymentOptions.ts

```diff
@@ -0,0 +1,186 @@
+    feePerByte: number,
+};
+
+export type NimiqDirectPaymentOptions = PaymentOptions<Currency.NIM, PaymentMethod.DIRECT>;
+
+export class ParsedNimiqDirectPaymentOptions extends ParsedPaymentOptions<Currency.NIM, PaymentMethod.DIRECT> {
+    public amount: number;
+    private extraData: Uint8Array;
+
+    public constructor(options: NimiqDirectPaymentOptions, extraData: Uint8Array) {
+        super(options);
+        this.extraData = extraData;
+        this.amount = Number.parseInt(toNonScientificNumberString(options.amount), 10);
+
+        let sender: Nimiq.Address | undefined;
+        if (options.protocolSpecific.sender !== undefined) {
+            try {
+                sender = Nimiq.Address.fromUserFriendlyAddress(options.protocolSpecific.sender);
+            } catch (err) {
+                throw new Error('If provided, sender must be a valid user friendly address string');
+            }
+        }
+
+        let recipient: Nimiq.Address | undefined;
+        if (options.protocolSpecific.recipient !== undefined) {
+            try {
+                recipient = Nimiq.Address.fromUserFriendlyAddress(options.protocolSpecific.recipient);
+            } catch (err) {
+                throw new Error('If provided, recipient must be a valid user friendly address string');
+            }
+        }
+
+        let recipientType: Nimiq.Account.Type | undefined;
+        if (options.protocolSpecific.recipientType !== undefined) {
+            if (!Object.values(Nimiq.Account.Type).includes(options.protocolSpecific.recipientType)) {
+                throw new Error('If provided, recipientType must be a valid nimiq account type');
+            }
+            recipientType = options.protocolSpecific.recipientType;
+        }
+
+        const flags = (options as NimiqDirectPaymentOptions).protocolSpecific.flags;
+        if (flags !== undefined && typeof flags !== 'number') {
+            throw new Error('If provided, flags must be a number.');
+        }
+
+        let feePerByte: number | undefined;
+        if (options.protocolSpecific.feePerByte !== undefined) {
+            try {
+                feePerByte = parseFloat(toNonScientificNumberString(options.protocolSpecific.feePerByte));
+            } catch (e) {
+                throw new Error('If provided, feePerByte must be a valid number');
+            }
+        }
+
+        let fee: number | undefined;
+        if (options.protocolSpecific.fee !== undefined) {
+            if (!this.isNonNegativeInteger(options.protocolSpecific.fee)) {
+                throw new Error('If provided, fee must be a non-negative integer');
+            }
+            fee = Number.parseInt(toNonScientificNumberString(options.protocolSpecific.fee), 10);
+        }
+
+        const requiresExtendedTransaction = extraData.length > 0
+            || recipientType !== undefined && recipientType !== Nimiq.Account.Type.BASIC
+            || flags !== undefined && flags !== Nimiq.Transaction.Flag.NONE;
```

Comment:
Do these conditions need brackets around them? Is `&&` stronger than `||`? Are you sure?